File: typescript/base-hmi-types/src/screens/shapes/HmiEllipseSegment.ts
import { HmiProperty, staticProperty } from "../base.js";
import { HmiEllipse } from "./HmiEllipse.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiEllipseSegment extends HmiEllipse {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiEllipseSegment;
  }

  startAngle: HmiProperty<number> = staticProperty(0);
  sweepAngle: HmiProperty<number> = staticProperty(0);
}

File: src/BaseHmiTypes/Screens/Shapes/HmiEllipseSegment.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiEllipseSegment : HmiEllipse
{
    public HmiEllipseSegment()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiEllipseSegment;
    }

    public HmiProperty<double> StartAngle { get; set; } = 0;

    public HmiProperty<double> SweepAngle { get; set; } = 0;
}
